content/docs/user-guide/interactivity/robotics/concepts-and-components-overview.md

```diff
@@ -0,0 +1,107 @@
+
+### Vehicle Model
+
+`VehicleModelComponent` serves the purpose of converting inputs such as target velocity, steering or acceleration to physical forces on parts of a vehicle (robot). `VehicleModel` has a `VehicleConfiguration` which is used to define axles, parametrize and assign wheels. The model requires a `WheelControllerComponent` present in each wheel entity. It also uses an implementation of `DriveModel`, which converts vehicle inputs to forces acting on steering elements and wheels.
+
+### Vehicle Dynamics
+
+See the [Vehicle Dynamics](vehicle-dynamics.md) section.
+
+### Spawner
+
+`ROS2SpawnerComponent` handles spawning entities during simulation. Available spawnables have to be set up as the component's field before the simulation. User is able to define named spawn points inside the Editor. This can be done by adding `ROS2SpawnPointComponent` to a child entity of an entity with `ROS2SpawnerComponent`. During the simulation user can access names of available spawnables and request spawning using ros2 services. The names of services are `/get_available_spawnable_names` and `/spawn_entity` respectivly. GetWorldProperties.srv and SpawnEntity.srv types are used to handle these features. In order to request defined spawn points names user can use `/get_spawn_points_names` service with `GetWorldProperties.srv` type. Detailed information about specific spawn point (e.g. pose) can be accessed using `/get_spawn_point_info` service with `GetModelState.srv` type. All used services types are defined in gazebo_msgs package.
+
+- **Spawning**: The spawnable name must be passed in `request.name` and the position of entity in `request.initial_pose`.
+  - Example call: 
+  ```
+  ros2 service call /spawn_entity gazebo_msgs/srv/SpawnEntity '{name: 'robot', initial_pose: {position:{ x: 4, y: 4, z: 0.2}, orientation: {x: 0.0, y: 0.0, z: 0.0, w:.0}}}
+  ```
+- Spawning in defined spawn point: spawnable name should be passed in request.name and the name of the spawn point in request.xml
+  - example call: `ros2 service call /spawn_entity gazebo_msgs/srv/SpawnEntity '{name: 'robot', xml: 'spawn_spot'}'`
+- Available spawnable names access: names of available spawnables are sent in response.model_names
+  - example call: `ros2 service call /get_available_spawnable_names gazebo_msgs/srv/GetWorldProperties`
+- Defined spawn points' names access: names of defined points are sent in response.model_names
+  - example call: `ros2 service call /get_spawn_points_names gazebo_msgs/srv/GetWorldProperties`
+- Detailed spawn point info access: spawn point name should be passed in request.model_name. Defined pose is sent in response.pose.
+  - example call: `ros2 service call /get_spawn_point_info gazebo_msgs/srv/GetModelState '{model_name: 'spawn_spot'}'`
```

Comment:
To fix formatting, similar to the above
```suggestion
- **Spawning in defined spawn point**: Pass in a spawnable into `request.name` and the name of the spawn point into `request.xml`.
  - Example call: 
    ``` 
    ros2 service call /spawn_entity gazebo_msgs/srv/SpawnEntity '{name: 'robot', xml: 'spawn_spot'}'
    ```
- **Available spawnable names access**: Send the names of available spawnables into `response.model_names`.
  - Example call: 
    ```
    ros2 service call /get_available_spawnable_names gazebo_msgs/srv/GetWorldProperties
    ```
- **Defined spawn points' names access**: Send the names of defined points into `response.model_names`
  - Example call: 
    ```
    ros2 service call /get_spawn_points_names gazebo_msgs/srv/GetWorldProperties
    ```
- **Detailed spawn point info access**: Pass in the spawn point name into `request.model_name` and the defined pose into `response.pose`.
  - Example call: 
    ```
    ros2 service call /get_spawn_point_info gazebo_msgs/srv/GetModelState '{model_name: 'spawn_spot'}'
    ```
```